### PR TITLE
Doc - Additional Deployment Refactor

### PR DESCRIPTION
**Pull request by sleto-it**
**Work in Progress - please do not merge**

Internal Ref:

- https://github.com/arangodb/planning/issues/2648
- https://github.com/arangodb/planning/issues/2649

### CODE REVIEW

**Comment by sleto-it** [comment]
@ewoutp Hi, when possible please could you do a first technical review on the following files of this PR - only the following files please:

- Documentation/Books/Manual/Deployment/Cloud/Aws.md
- Documentation/Books/Manual/Deployment/Cloud/Azure.md
- Documentation/Books/Manual/Deployment/README.md

Any comment from @neunhoef also welcome. We can discuss more in a meeting.

This PR is yet another effort in the direction of simplifying the way we  document deployment, first step for a successful experience with ArangoDB. In particular, here we will focus on:

1. Removing **old and outdated** content related to deployment of ArangoDB on Cloud
2. Aggregate (and hopefully review) all the info that we currently have on DC/OS

First focus is on Cloud, and second on DC/OS. This is still a WIP

Many thanks,

File: Documentation/Books/Cookbook/Cloud/GiantSwarm.md [hunk 1]
```diff
@@ -1,197 +0,0 @@
...
-The important line is
-
-    "scaling_policy": { "min": 2, "max": 2 }
-    
-It tells the swarm to use two front-end containers. In later version of the `swarm` you will be able to change the number of containers in a running application with the command:
-
-    > swarm scaleup guesser/guesser-game/guesser-front-end --count=1
-    Scaling up component guesser/guesser-game/guesser-front-end by 1...
-
-We at ArangoDB are hard at work to make scaling up the back-end database equally easy. Stay tuned for new releases in early 2015...
```
Comment (ewoutp) [inline Documentation/Books/Cookbook/Cloud/GiantSwarm.md]: Really outdated

File: Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+Cluster Deployments in Apache Mesos
+===================================
+
+ArangoDB has a sophisticated and yet easy way to use cluster mode. To leverage the full cluster feature set (monitoring, scaling, automatic failover and automatic replacement of failed nodes) you have to run ArangoDB on some kind of cluster management system. Currently ArangoDB relies on Apache Mesos in that matter. Mesos is a cluster operating system which powers some of the worlds biggest datacenters running several thousands of nodes.
```
Comment (ewoutp) [inline Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md]: This sounds like we only support mesos and recommend it. That is no longer correct.

File: Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    "--nr_coordinators=2",
+    "--failover_timeout=86400",
+    "--arangodb_image=arangodb/arangodb-mesos:3.1",
+    "--secondaries_with_dbservers=false",
+    "--coordinators_with_dbservers=false"
+  ],
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "arangodb/arangodb-mesos-framework:3.1",
```
Comment (ewoutp) [inline Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md]: 3.1: Is that still correct?

File: Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    "--async_replication=true",
+    "--minimal_resources_agent=mem(*):512;cpus(*):0.25;disk(*):512",
+    "--minimal_resources_dbserver=mem(*):512;cpus(*):0.25;disk(*):1024",
+    "--minimal_resources_secondary=mem(*):512;cpus(*):0.25;disk(*):1024",
+    "--minimal_resources_coordinator=mem(*):512;cpus(*):0.25;disk(*):1024",
+    "--nr_agents=1",
+    "--nr_dbservers=2",
+    "--nr_coordinators=2",
+    "--failover_timeout=86400",
+    "--arangodb_image=arangodb/arangodb-mesos:3.1",
```
Comment (ewoutp) [inline Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md]: 3.1: Is that still correct

File: Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+### Deinstallation via Marathon
+
+As with DC/OS you first need to properly cleanup any state leftovers.
+
+The easiest is to simply delete ArangoDB and then deploy the cleanup-framework (see section arangodb-cleanup-framework).
+
+Configuration options
+---------------------
+
+The Arangodb Mesos framework has a ton of different options which are listed and described here: https://github.com/arangodb/arangodb-mesos-framework/tree/3.2
```
Comment (ewoutp) [inline Documentation/Books/Manual/Deployment/DCOS/ClusterMesos.md]: Before 3.1 was used, now 3.2. Which is correct?

File: Documentation/Books/Manual/Deployment/README.md [hunk 6]
```diff
@@ -21,3 +21,8 @@ By _Technology_:
 - [Docker](Docker/README.md)
 - [Kubernetes](Kubernetes/README.md)
 - [Mesos, DC/OS](DCOS/README.md)
+
+By _Cloud Provider_:
```
Comment (ewoutp) [inline Documentation/Books/Manual/Deployment/README.md]: IMO "By" is the wrong word here. 
We already have "By deployment" & "By technology" and both are still applicable when you go in the cloud. The topics described under here are  purely an intro and referral to various "By technology" items.

So consider calling it "In the cloud"

File: Documentation/Books/Manual/SUMMARY.md [hunk 7]
```diff
@@ -247,6 +247,13 @@
       * [TLS](Deployment/Kubernetes/Tls.md)
       * [Troubleshooting](Deployment/Kubernetes/Troubleshooting.md)
     * [Mesos, DC/OS](Deployment/DCOS/README.md)
+      * [Cluster Deployments](Deployment/DCOS/ClusterMesos.md)
+      * [ArangoDB in Mesos](Deployment/DCOS/Mesosphere.md)
+      * [DC/OS: Full example](Deployment/DCOS/DcosFullExample.md)
+      * [DC/OS: Choosing Container engine](Deployment/DCOS/MesosContainers.md)
+    * [By Cloud Provider](Deployment/Cloud/README.md)
```
Comment (ewoutp) [inline Documentation/Books/Manual/SUMMARY.md]: Same "By" comment as above

**Pull request merged**